common/src/main/java/com/redhat/cloud/notifications/models/InstantEmailTemplate.java

```diff
@@ -0,0 +1,178 @@
+package com.redhat.cloud.notifications.models;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.redhat.cloud.notifications.models.filter.ApiResponseFilter;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+import java.util.UUID;
+
+import static com.fasterxml.jackson.annotation.JsonProperty.Access.READ_ONLY;
+import static com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
+import static javax.persistence.FetchType.LAZY;
+
+// TODO NOTIF-484 Add templates ownership and restrict access to the templates.
+@Entity
+@Table(name = "instant_email_template")
+@JsonNaming(SnakeCaseStrategy.class)
+@JsonFilter(ApiResponseFilter.NAME)
+public class InstantEmailTemplate extends CreationUpdateTimestamped {
+
+    @Id
+    @GeneratedValue
+    @JsonProperty(access = READ_ONLY)
+    private UUID id;
+
+    @ManyToOne
+    @JoinColumn(name = "event_type_id")
+    private EventType eventType;
+
+    @Transient
+    private UUID eventTypeId;
+
+    @ManyToOne(fetch = LAZY, optional = false)
+    @JoinColumn(name = "subject_template_id")
+    @JsonProperty(access = READ_ONLY)
+    private Template subjectTemplate;
+
+    @NotNull
+    @Transient
+    private UUID subjectTemplateId;
```

Comment:
are these transient fields actually used? or could be replaced by a getter ? (I think jackson picks the fields from `getStuff` fields when serializing) - Not specific to this PR, but maybe we could refactor to remove these extra fields.

Reply:
Jackson serializes fields by default AFAIK.

About this specific field, yes it is used in one of the REST responses.

Reply:
The point i tried to convey is that (IIUC) this `subjectTemplateId` is a synthethic field that we only add for the REST responses.
Removing this field and creating a method:
```
@JsonProperty
getSubjectTemplateId() {
  return subjectTemplate.getId()
}
```

would yield the same behavior without the extra field that we need to keep in sync with the `subjectTemplate.id`

Reply:
We could do that indeed.

Reply:
I forgot that we need that field for the incoming REST request. 😄

Reply:
Lets explore alternatives for that in the future (and outside of this PR)